apps/sla/models.py

```diff
@@ -24,39 +39,50 @@ class SLA:
         "latest": max,
     }
 
-    def __init__(self, sla_desc):
-        self.duration = int(sla_desc["duration"])
-        self.add_days = self.ADD_DAYS[sla_desc["type"]]
-        self._init_start(sla_desc["start"])
-
-    def _init_start(self, start_desc):
-        """
-        SLA start obtaining description parsing
-        """
-        # allow the parsering of the shorter
-        # style definitions the same way
-        if isinstance(start_desc, str):
-            start_desc = {"latest": [start_desc]}
+    duration = models.IntegerField()
+    duration_type = models.CharField(max_length=20, choices=DURATION_TYPES)
+    start_criteria = models.CharField(max_length=20, choices=START_CHOICES)
+    start_dates = fields.ArrayField(
+        models.CharField(max_length=100), default=list, null=True
```

Comment:
Is it expected that `start_dates` can be NULL (None)? It might be sufficient to go with just `blank=True` and always count on empty list.

Reply:
A valid policy must have some start date defined (not saying that it must be necessarily required here).

Reply:
Actually if the SLA is well defined it should neither be blank or null, so I'll just remove the `null=True`